### PR TITLE
[libvpx,lmdb,aubio,freetype-gl,intelrdfpmathlib,libbson,libtcod,metis,pqp,smpeg2] Build fixes 2022-04-28

### PR DESCRIPTION
**Pull request by BillyONeal**
These results are from the most recent CI run: https://dev.azure.com/vcpkg/public/_build/results?buildId=71465

PASSING, REMOVE FROM FAIL LIST: aubio:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).

I also did some investigation as to why aubio:arm-uwp didn't pass. Turns out, it's because aubio depends on ffmpeg, which failed to build because it depends on libvpx, which we never fixed for UWP following the VS2022 update. See also https://developercommunity.visualstudio.com/t/MicrosoftVisualStudioComponentVCTool/10002207?space=62&scope=follow&sort=newest

PASSING, REMOVE FROM FAIL LIST: freetype-gl:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).

I also checked freetype-gl:arm-uwp, but it's blocked by glew which is blocked by opengl which appears to not be a thing on arm.

PASSING, REMOVE FROM FAIL LIST: intelrdfpmathlib:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).

The arm-uwp version of this emits errors that look like source issues; I blocked arm&windows with a supports expression:
D:\buildtrees\intelrdfpmathlib\src\athLib20U2-d2a8954428.clean\LIBRARY\src\bid_functions.h(3113): error C2719: 'x': formal parameter with requested alignment of 16 won't be aligned

PASSING, REMOVE FROM FAIL LIST: libbson:arm-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: libbson:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: libtcod:arm-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: libtcod:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: lmdb:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).

arm-uwp failed with again what looks like a source issue:
mdb.c.obj : error LNK2001: unresolved external symbol __tls_used
mdb.c.obj : error LNK2001: unresolved external symbol _mdb_tls_cbp

PASSING, REMOVE FROM FAIL LIST: metis:arm-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: metis:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: pqp:arm-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: pqp:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: smpeg2:arm-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
PASSING, REMOVE FROM FAIL LIST: smpeg2:x64-uwp (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).

I also checked Linux and it says
  Could NOT find ibverbs (missing: IBVERBS_INCLUDE_DIRS IBVERBS_LIBRARIES)
which may be vcpkg's fault so I left that ci.baseline.txt skip alone.

REGRESSION: jansson:arm-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add jansson:arm-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jansson:x64-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add jansson:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.

Already fixed by https://github.com/microsoft/vcpkg/pull/24466

THANK YOU THANK YOU THANK YOU @Neumann-A for fixing all these.

/cc @cenit this fixes vcpkg-ci-ffmpeg:arm-uwp

### CODE REVIEW

**Comment by cenit** [comment]
so it turns out msbuild is a real limit for uwp??

**Comment by BillyONeal** [comment]
> so it turns out msbuild is a real limit for uwp??

My guess is that it's those warnings we couldn't figure out how to disable.

File: ports/lmdb/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "lmdb",
   "version": "0.9.29",
-  "port-version": 1,
+  "port-version": 2,
   "description": "LMDB is an extraordinarily fast, memory-efficient database",
   "homepage": "https://github.com/LMDB/lmdb",
   "license": null,
+  "supports": "!(arm & windows)",
```
Comment (ras0219-msft) [inline ports/lmdb/vcpkg.json:8]: Should this be `arm & uwp`, instead of all `windows`?

**Review by ras0219-msft — APPROVED** [review]

File: ports/lmdb/vcpkg.json (hunk 1 above)
Comment (BillyONeal) [reply ports/lmdb/vcpkg.json:8]: The compiler errors were about broken intrinsics that seem unlikely to be related to UWP.

**Pull request merged**